lib/matplotlib/rcsetup.py

```diff
@@ -601,16 +601,16 @@ def validate_markevery(s):
 validate_legend_loc = ValidateInStrings(
     'legend_loc',
     ['best',
-     'upper right',
-     'upper left',
-     'lower left',
-     'lower right',
+     'upper right', 'NE', 'northeast',
+     'upper left', 'NW', 'northwest',
+     'lower left', 'SW', 'southwest',
+     'lower right', 'SE', 'southeast',
      'right',
-     'center left',
-     'center right',
-     'lower center',
-     'upper center',
-     'center'], ignorecase=True)
+     'center left', 'W', 'west',
+     'center right', 'E', 'east',
+     'lower center', 'S', 'south',
+     'upper center', 'N', 'north',
+     'center', 'C'], ignorecase=True)
```

Comment:
Add comment that `ignorecase=False` after a deprecation period (i.e. in 3.3).

Reply:
This may be an important point. This PR does not actually deprecate the use of other casing *for rcParams*. If this is desired I would need to change this now.

Reply:
You will get a deprecation notice when the user for a rcParam value 'Upper Center' when that value is applied, won't you? So it's effectively deprecated. It's only at the end of the deprecation period that the rcParam checker can get more strict and check the case.

Reply:
No, actually, you won't get any warning for rcParameters here because with `ignorecase=True`, they are normalized to the `valid` parameters. 

```
from matplotlib.rcsetup import ValidateInStrings

v = ValidateInStrings("cattle", ["cow", "SHEEP"], ignorecase=True)

print(v("Cow"))     # returns 'cow', no warning
print(v("sheep"))   # returns 'SHEEP', no warning
print(v("Hamster")) # ValueError: Unrecognized cattle string 'hamster'
```

I'm currently thinking about directly using the newly introduced `_map_loc_to_compass` directly for rc validation.

Reply:
Sounds reasonable.

Reply:
As of now rcParam "legend.loc" only accepts strings (numbers or tuples are not allowed). Is there intention to change that, such that those are allowed as rcParams as well? 

I'm asking since with that parameter running through `_map_loc_to_compass`, 

    plt.rcParams["legend.loc"] = 2

would be allowed, but the same from an rc file

    legend.loc : 2

would fail due to missing string to int conversion - which I could add of course if that is desired.